tests/partitions_test.py

```diff
@@ -0,0 +1,104 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache 2.0 License.
+
+import infra.network
+import infra.e2e_args
+import infra.partitions
+import infra.logging_app as app
+import suite.test_requirements as reqs
+
+from loguru import logger as LOG
+
+
+@reqs.description("Invalid partitions are not allowed")
+def test_invalid_partitions(network, args):
+    nodes = network.get_joined_nodes()
+
+    try:
+        network.partitioner.partition(
+            [nodes[0], nodes[2]],
+            [nodes[1], nodes[2]],
+        )
+        assert False, "Node should not appear in two or more partitions"
+    except ValueError:
+        pass
+
+    try:
+        network.partitioner.partition()
+        assert False, "At least one partition should be specified"
+    except ValueError:
+        pass
+
+    try:
+        invalid_local_node_id = -1
+        new_node = infra.node.Node(invalid_local_node_id, "local://localhost")
+        network.partitioner.partition([new_node])
+        assert False, "All nodes should belong to network"
+    except ValueError:
+        pass
+
+    return network
+
+
+@reqs.description("Partition primary + f nodes")
+def test_partition_majority(network, args):
+    primary, backups = network.find_nodes()
+
+    # Create a partition with primary + half remaining nodes (i.e. majority)
+    partition = [backups[0]]
+    partition.extend(backups[len(backups) // 2 :])
```

Comment:
What am I missing here?

It looks like this creates a majority partition that doesn't include the primary (it includes the first backup, and the last half-rounded-up backups), which should cause an election?

Reply:
Good spot. Changed to `partiton = [primary]` to match the comment and intent of the test.